### PR TITLE
Add Psalm

### PR DESCRIPTION
**Pull request by nickvergessen**
Signed-off-by: Joas Schilling <coding@schilljs.com>

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -2,9 +2,14 @@
...
+		"lint": "find . -name \\*.php -not -path './vendor/*' -not -path './build/*' -not -path './tests/integration/vendor/*' -print0 | xargs -0 -n1 php -l",
+		"psalm": "psalm",
+		"psalm:fix": "psalm --alter --issues=InvalidReturnType,InvalidNullableReturnType,MissingParamType,InvalidFalsableReturnType"
 	},
+	"minimum-stability": "dev",
 	"require-dev": {
-		"nextcloud/coding-standard": "^0.3.0"
+		"nextcloud/coding-standard": "^0.3.0",
+		"vimeo/psalm": "^3.16",
+		"christophwurst/nextcloud": "dev-master"
```
Comment (ChristophWurst) [inline composer.json:13]: you can drop it here and only install on CI

Comment (nickvergessen) [reply composer.json:13]: but then one can not run it locally?
Or is it recomment to install this and psalm locally manually?

Comment (ChristophWurst) [reply composer.json:13]: Yeah, right. You can leave it then. Or install the OCP package only on demand when you want to psalm locally. I'd keep psalm definitely in composer.json as that locks the version.

**Comment by PVince81** [comment]
seems it fails now on master https://github.com/nextcloud/spreed/issues/4404

**Pull request merged**